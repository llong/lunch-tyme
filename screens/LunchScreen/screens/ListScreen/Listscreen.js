import React from 'react';
import { connect } from 'react-redux';
import { TouchableOpacity, FlatList, View, Dimensions, Image } from 'react-native';
import { bindActionCreators } from 'redux';
import ListItem from './components/ListItem';
import { setActiveRestaurant } from '../../../../state/actions';
import MapIcon from '../../../../assets/icons/icon_map.png';
const deviceWidth = Dimensions.get('window').width;

class ListScreen extends React.Component {
  static navigationOptions = ({navigation}) => {
    return {
      title: 'Lunch Tyme',
      headerRight: (
        <Image source={ MapIcon } style={ { marginRight: 12 } } />
      )
    };
  }

  constructor(props) {
    super(props);
    this.state = {
      restaurants: []
    }
  }

  componentDidMount() {
    const restaurants = fetch('http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/restaurants.json')
      .then(res => {
        if(res.status !== 200) {
          alert('problem fetching');
          return
        }
        res.json().then( data => this.setState({ restaurants: data.restaurants }));
      })
      .catch(err => {
        alert('there was a problem ', err);
      })
  }

  navigateToDetails = async (item) => {
    const { navigation, setActiveRestaurant: doSetActiveRestaurant } = this.props;
    await doSetActiveRestaurant(item);
    navigation.navigate('Details');
  }

  renderItem = ({item}) => {
    return (
      <TouchableOpacity onPress={() => this.navigateToDetails(item)}>
        <ListItem item={ item } />
      </TouchableOpacity>
    );
  }
  
  keyExtractor = (item, index) => item.name;

  renderRestaurants = () => {
    const { restaurants } = this.state;
    return (
      <FlatList
        contentContainerStyle={{ width: deviceWidth, backgroundColor: '#000000'}}
        data={ restaurants }
        renderItem={ this.renderItem }
        keyExtractor={ this.keyExtractor }
        bounces={ false }
      />
    )
  }

  render() {
    const { restaurants } = this.state;
    return (
      <View style={{ flex: 1 }}>
        { restaurants.length >= 1 && this.renderRestaurants() }
      </View>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    setActiveRestaurant
  }, dispatch)
}

export default connect(null, mapDispatchToProps)(ListScreen);
