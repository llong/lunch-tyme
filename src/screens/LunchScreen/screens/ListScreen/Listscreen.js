// @flow

import React from 'react';
import { connect } from 'react-redux';
import {
  TouchableOpacity,
  FlatList,
  View,
  Dimensions,
  Image,
  Alert,
} from 'react-native';
import { bindActionCreators } from 'redux';
import ListItem from './components/ListItem';
import { setActiveRestaurant } from '../../../../state/actions';
import MapIcon from '../../../../assets/icons/icon_map.png';
import styles from './styles';

const deviceWidth = Dimensions.get('window').width;

type Props = {
  navigation: {
    navigate: (string) => mixed
  },
  setActiveRestaurant: (Object) => Object
}
type State = {
  restaurants: Array<Object>
}

class ListScreen extends React.Component<Props, State> {
  static navigationOptions = {
    title: 'Lunch Tyme',
    headerRight: (
      <Image source={MapIcon} style={styles.headerIcon} />
    ),
  }

  constructor(props) {
    super(props);
    this.state = {
      restaurants: [],
    };
  }

  componentDidMount() {
    fetch('http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/restaurants.json')
      .then((res) => {
        if (res.status !== 200) {
          Alert('problem fetching');
          return;
        }
        res.json().then(data => this.setState({ restaurants: data.restaurants }));
      })
      .catch((err) => {
        Alert('there was a problem ', err);
      });
  }

  navigateToDetails = async (item) => {
    const { navigation, setActiveRestaurant: doSetActiveRestaurant } = this.props;
    await doSetActiveRestaurant(item);
    navigation.navigate('Details');
  }

  renderItem = ({ item }) => (
    <TouchableOpacity onPress={() => this.navigateToDetails(item)}>
      <ListItem item={item} />
    </TouchableOpacity>
  );

  keyExtractor = item => item.name;

  renderRestaurants = () => {
    const { restaurants } = this.state;
    return (
      <FlatList
        contentContainerStyle={{ width: deviceWidth, backgroundColor: '#000000' }}
        data={restaurants}
        renderItem={this.renderItem}
        keyExtractor={this.keyExtractor}
        bounces={false}
      />
    );
  }

  render() {
    const { restaurants } = this.state;
    return (
      <View style={{ flex: 1 }}>
        { restaurants.length >= 1 && this.renderRestaurants() }
      </View>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    setActiveRestaurant,
  }, dispatch);
}

export default connect(null, mapDispatchToProps)(ListScreen);
