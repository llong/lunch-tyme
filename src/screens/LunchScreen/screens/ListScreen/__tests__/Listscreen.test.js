import React from 'react';
import { shallow } from 'enzyme';
import { ListScreen } from '../Listscreen';

const navigateMock = jest.fn();

const props = {
  navigation: {
    navigate: navigateMock,
  },
};

describe('ListScreen', () => {
  const wrapper = shallow(<ListScreen {...props} />, { disableLifecycleMethods: true });
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
