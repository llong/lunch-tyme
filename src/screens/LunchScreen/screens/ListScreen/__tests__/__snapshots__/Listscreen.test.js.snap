// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListScreen should render component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <FlatList
    bounces={false}
    contentContainerStyle={
      Object {
        "backgroundColor": "#000000",
        "width": 750,
      }
    }
    data={
      Array [
        Object {
          "category": "fast food",
          "name": "McDonalds",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
